depthai_helpers/managers.py

```diff
@@ -551,46 +552,43 @@ class FPSHandler:
     fps_line_type = cv2.LINE_AA
 
     def __init__(self, cap=None):
-        self.timestamp = time.monotonic()
+        self.timestamp = None
         self.start = None
         self.framerate = cap.get(cv2.CAP_PROP_FPS) if cap is not None else None
         self.useCamera = cap is None
 
-        self.frame_cnt = 0
+        self.iter_cnt = 0
         self.ticks = {}
-        self.ticks_cnt = {}
 
     def next_iter(self):
         if self.start is None:
             self.start = time.monotonic()
 
-        if not self.useCamera:
+        if not self.useCamera and self.timestamp is not None:
             frame_delay = 1.0 / self.framerate
             delay = (self.timestamp + frame_delay) - time.monotonic()
             if delay > 0:
                 time.sleep(delay)
         self.timestamp = time.monotonic()
-        self.frame_cnt += 1
+        self.iter_cnt += 1
 
     def tick(self, name):
-        if name in self.ticks:
-            self.ticks_cnt[name] += 1
-        else:
-            self.ticks[name] = time.monotonic()
-            self.ticks_cnt[name] = 0
+        if name not in self.ticks:
+            self.ticks[name] = collections.deque(maxlen=10)
```

Comment:
The default queue length of 10 is a bit small, and doesn't filter well the jitter. I'd suggest about 30, or maybe even better like 100 (a max window of 3.33s at 30 fps).
Ideally the averaging window length in time would be runtime-adjusted based on the FPS, still having a max-sized `deque`, but taking only the last `N` samples in the FPS calc, with 
`N = math.ceil(FPS / predefined_window_length_seconds) + 1`.
But I think it's fine to just increase the max window length for now.

Reply:
Thanks! yes, agreed, I think 100 will suit well for now, committed already